### DISCUSSION TITLE
dynamic page in root of pages directory

### DISCUSSION THREAD

**Discussion by ryan-reach** [vercel/next.js#12255] [Help] [answered]
We're fetching page data from a headless cms so all our pages use `getServerSideProps`. Our `pages` directory looks like this:
```
pages/
  index.js
  news.js
  about.js
  [custom].js
  api/
    email.js
```
We have a dynamic route for custom pages created by the cms editors. When I refresh or server-side route to the home page, news, or about, `getServerSideProps` in `[custom].js` is also called.

Even stranger, when I log the context.params in `[custom].js` I get `{ custom: 'sw.js' }`. I'm not too familiar with service workers, but I'm guessing that's what that is...

We basically want to use `[custom].js` as a catch all route, I tried `[...custom].js` but it didn't make a difference. I'm guessing having a dynamic page in the root of pages is just a bad idea? Hoping I can find a solution that doesn't involve a custom server..

on nextjs 9.3.3

**Comment by timneutkens** [comment, marked as answer]
It's probably requesting `favicon.ico` which hits `[custom].js` if you don't have a favicon.ico in your project.

**Reply by ryan-reach** [reply]
hmm, we're fetching a png from the cms to use as the favicon. I did just try adding a favicon to the public directory, but `getServerSideProps` in `[custom].js` is still being called with `sw.js` as the url param. very odd...

**Reply by timneutkens** [reply]
That's a serviceworker that you still have installed it seems

**Reply by ryan-reach** [reply]
yeah, I wonder if it could it be coming from a npm package? we don't have sw.js in our project... I appreciate your quick response. i'll keep digging

**Reply by timneutkens** [reply]
What i meant to say is that you probably previously had a serviceworker installed and it's trying to request that 🤔

**Reply by ryan-reach** [reply]
huh, interesting. I tried deleting my `.next` directory and re-running the project but that didn't seem to make a difference. It's not breaking anything in development, I haven't tried deploying it yet

**Reply by rafaelalmeidatk** [reply]
If I am not mistaken, you could also have installed a service worker with another project that was running in the same port of your Next project.

You can unregister services workers going to DevTools > Application > Service Workers > Unregister

**Reply by ryan-reach** [reply]
@rafaelalmeidatk that was totally it! thank you guys so much for your help!
